src/main/java/frc/robot/subsystems/BallTransport.java

```diff
@@ -220,18 +223,31 @@ public CommandBase removeCargo() {
                     topMotor.set(REMOVE_SPEED);
                     bottomMotor.set(REMOVE_SPEED);
                     colorBuffer.clear();
+                    topFlag = true;
+                    bottomFlag = true;
                 }
             } else {
                 if (colorBuffer.peekLast() == oppositeAllianceBallColor) {
                     topMotor.set(REMOVE_SPEED);
                     colorBuffer.removeLast();
+                    bottomFlag = true;
                 }
                 if (colorBuffer.peekFirst() == oppositeAllianceBallColor) {
                     bottomMotor.set(-REMOVE_SPEED);
                     colorBuffer.removeFirst();
+                    topFlag = true;
                 }
             }
-        }).onEnd(this::stop);
+        }).until(() -> {
+            if (bottomFlag && topFlag) {
```

Comment:
This entire block can be simplified:

- If `bottomFlag` is set, then `colorSensorBallLimit()` is part of the result. In other words, `bottomFlag && colorSensorBallLimit()`
- If `topFlag` is set, then `!laserSwitch.getAsBoolean()` is part of the result. In other words, `topFlag && !laserSwitch.getAsBoolean()`.
- If neither is set, then the result is false.

Put all these together and you get `(bottomFlag && colorSensorBallLimit()) && (topFlag && !laserSwitch.getAsBoolean())`, and the whole if-else structure goes away.